### DISCUSSION TITLE
how i can convert this code to  `export default function`

### DISCUSSION THREAD

**Discussion by hajpepe** [vercel/next.js#38065] [Help] [answered]
```
import Link from 'next/link';
import { useState } from 'react';
import Logo from '../constants/svg/logo.svg';

export const Navbar = () => {
  const [active, setActive] = useState(false);

  const handleClick = () => {
    setActive(!active);
  };

  return (
    <>
      <nav className='sticky top-0 z-40 flex flex-wrap items-center p-3 bg-white drop-shadow-md '>
        <Link href='https://app.xpeace.cloud/login'>
          <button className="hidden ml-[35px]   bg-[#E9E9E9] hover:bg-sky-700 items-center justify-center w-full px-3 py-2 font-bold text-black rounded lg:inline-flex w-24 hover:bg-[#b4b4b4] hover:text-black ">
            ورود
          </button>
        </Link>
        <Link href='https://app.xpeace.cloud/register'>
          <button className="hidden  ml-[20px] bg-[#1361FF] hover:bg-sky-700 items-center justify-center w-full px-3 py-2 font-bold text-white rounded lg:inline-flex w-32 hover:bg-[#0f4fce] hover:text-black">
            ثبت نام
          </button>
        </Link>
        <button
          className='inline-flex p-3 mr-auto  text-black rounded outline-none hover:bg-[#1361FF] lg:hidden hover:text-white'
          onClick={handleClick}
        >
          <svg className='w-6 h-6' fill='none' stroke='currentColor' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'><path strokeLinecap='round' strokeLinejoin='round' strokeWidth={2} d='M4 6h16M4 12h16M4 18h16' /></svg>
        </button>
        {/*Note that in this div we will use a ternary operator to decide whether or not to display the content of the div  */}
        <div
          className={`${active ? '' : 'hidden'
            }   w-full lg:inline-flex lg:flex-grow lg:w-auto`}
        >
          <div className='flex flex-col items-start w-full lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto lg:items-center lg:h-auto'>
            <Link href='/about'>
              <a className='items-center justify-center w-full px-3 py-2 font-bold text-center text-black rounded lg:inline-flex lg:w-auto hover:text-bluext '>
                درباره ما
              </a>
            </Link>
            <Link href='/'>
              <a className='items-center justify-center w-full px-3 py-2 font-bold text-center text-black rounded lg:inline-flex lg:w-auto hover:text-bluext '>
                پشتیبانی
              </a>
            </Link>
            <Link href='/about'>
              <a className='items-center justify-center w-full px-3 py-2 font-bold text-center text-black rounded lg:inline-flex lg:w-auto hover:text-bluext '>
                تعرفه ها
              </a>
            </Link>
            <Link href='/'>
              <a className='items-center justify-center w-full px-3 py-2 font-bold text-center text-black rounded lg:inline-flex lg:w-auto hover:text-bluext '>
                خانه
              </a>
            </Link>
          </div>
        </div>
        <Link href='/'>
          <a className='inline-flex items-center p-2 mr-4 '>
            <Logo className='text center cinema' />
          </a>
        </Link>
      </nav>
    </>
  );
};
```
how i can convert this code to  `export default function`

**Comment by alex289** [comment, marked as answer]
Replace "export const Navbar = () => {" with "export default function Navbar() {"

But this would change nothing in the behavior of the code.

**Reply by hajpepe** [reply]
Server Error
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
:D

**Reply by hajpepe** [reply]
before i tested but not work!

**Reply by alex289** [reply]
How do you import and use this component?

**Reply by hajpepe** [reply]
`import { Navbar } from "./components/navbar"; `
```
export default function Home() {
  return (
    <div>
      <Navbar />

      <Footer />
    </div>
  );
}
```

**Reply by hajpepe** [reply]
`import Navbar from "./components/navbar";`  ( i tested this before )

**Reply by alex289** [reply]
> `import { Navbar } from "./components/navbar"; `
> 
> ```
> 
> export default function Home() {
> 
>   return (
> 
>     <div>
> 
>       <Navbar />
> 
> 
> 
>       <Footer />
> 
>     </div>
> 
>   );
> 
> }
> 
> ```

This will only work for named exports (without default).

**Reply by alex289** [reply]
> `import Navbar from "./components/navbar";`  ( i tested this before )

This should work with the default export tho. I really don't know why it's not working but I guess something else is not working.

Maybe it's the svg which can't be loaded.

**Reply by hajpepe** [reply]
```
import Link from 'next/link';
import { useState } from 'react';
import Logo from '../constants/svg/logo.svg';

export const Navbar = () => {
  const [active, setActive] = useState(false);

  const handleClick = () => {
    setActive(!active);
  }; 

  return (
```

i think this is problem 

```
import { useState } from 'react';

  const [active, setActive] = useState(false);

  const handleClick = () => {
    setActive(!active);
  }; 
```

**Reply by alex289** [reply]
No this should work. The error has something to do with an import problem. My guess is: the svg can't be imported like this because as the other comment says you don't have a loader for svgs in development and nextjs can't import non exported things

**Reply by hajpepe** [reply]
dude u dont see
`import Logo from '../constants/svg/logo.svg';`
? svgr

**Reply by hajpepe** [reply]
i deleted svg nothing change

**Reply by hajpepe** [reply]
i fixed thats problem

**Comment by icyJoseph** [comment]
Do you have any loader for that SVG?

I think, depending on your Next.js version various errors might come up on this one. I'd recommend using this loader, https://react-svgr.com/docs/next/ - all you gotta do is install it, and add the webpack configuration they indicate. Then you are good to go.

**Reply by hajpepe** [reply]
> Do you have any loader for that SVG?
> 
> I think, depending on your Next.js version various errors might come up on this one. I'd recommend using this loader, https://react-svgr.com/docs/next/ - all you gotta do is install it, and add the webpack configuration they indicate. Then you are good to go.

im using svgr but not in development first copy/paste svg and in end of code using svgr

**Reply by icyJoseph** [reply]
Yeah, I do that too, the proper you have it right, but without an SVG loader, I think an object is loaded instead, and that's causing this issue. Do you have a link to a repository? I am afraid there's more to this issue than we can see here.

Btw, using the SVG raw like this, is something I've come to love, our designers can keep on working with their svg's and we just import them as React components, how wonderful is that. Of course, a proper loader is needed for the extension `.svg`.

> im using svgr but not in development first copy/paste svg and in end of code using svgr

I did not understand what you mean here? You have installed `svgr`, and have added the WebPack configuration for it, right?

**Comment by nanataly** [comment]
Thank you for sharing. it is very useful and easy to follow.
Have a good day!
@my friend [Geometry Dash Unblocked](https://geometrydashunblocked.io)
